dfusion_rust_core/src/models/order.rs

```diff
@@ -32,9 +43,46 @@ impl Serializable for u128 {
     }
 }
 
+impl From<Arc<Log>> for Order {
+    fn from(log: Arc<Log>) -> Self {
+        let mut bytes: Vec<u8> = log.data.0.clone();
```

Comment:
ubernit: we shouldn't need to clone here. But can fix that in a forward PR given that this one already passed CI.

Reply:
sounds good, merging

Reply:
Maybe a few more words on when we need to clone, since I also saw it in @anxolin's PR.

Clone makes a copy of the variable and leaves the existing instance untouched and thus "consumable". Whenever the original variable is not used later, this is unnecessary since we could have also "consumed" (aka moved) it.
If the variable is used later we might have to use clone, but only if the current use of it is "consuming". That is, e.g. an assignment or if the variable passed by value into another function (if we wrote the function it can be worthwhile checking if it has to be passed by value).
If we can use the value by reference (e.g. in a print statement), we don't consume the variable and thus doesn't require a clone (just an & before the variable).